Assignment1/irisdata.py

```diff
@@ -0,0 +1,246 @@
+    # scatter plot of sepal length and sepal width
+    fig = px.scatter(
+        data_df,
+        x="sepal.width",
+        y="sepal.length",
+        color="species",
+        symbol="species",
+        title="Scatter plot of Seoal Length & Sepal Width",
+    )
+    fig.show()
+    # scatter plot of petal length and petal width
+    fig = px.scatter(
+        data_df,
+        x="petal.width",
+        y="petal.length",
+        color="species",
+        symbol="species",
+        title="Scatter Plot of Petal Length & Petal Width",
+    )
+    fig.show()
+    # Violin and Box Plots
+    fig1 = px.violin(
+        data_df, y="sepal.length", box=True, title="Violin & Box Plot of Sepal Length"
+    )
+    fig2 = px.violin(
+        data_df, y="sepal.width", box=True, title="Violin & Box Plot of Sepal Width"
+    )
+    fig3 = px.violin(
+        data_df, y="petal.length", box=True, title="Violin & Box Plot Petal Length"
+    )
+    fig4 = px.violin(
+        data_df, y="petal.width", box=True, title="Violin & Box Plot Petal Width"
+    )
+    fig1.show()
+    fig2.show()
+    fig3.show()
+    fig4.show()
+    # Bar Plots
+    fig5 = px.bar(
+        data_df,
+        x="species",
+        y="sepal.length",
+        color="species",
+        title="Bar Plot of Sepal Length",
+    )
+    fig6 = px.bar(
+        data_df,
+        x="species",
+        y="sepal.width",
+        color="species",
+        title="Bar Plot of Sepal width",
+    )
+    fig7 = px.bar(
+        data_df,
+        x="species",
+        y="petal.length",
+        color="species",
+        title="Bar Plot of Petal Length",
+    )
+    fig8 = px.bar(
+        data_df,
+        x="species",
+        y="petal.width",
+        color="species",
+        title="Bar Plot of Petal Width",
+    )
+    fig5.show()
+    fig6.show()
+    fig7.show()
+    fig8.show()
+    # Histogram
+    fig9 = px.histogram(
+        data_df, x="sepal.length", color="species", title="Histogram of Sepal length"
+    )
+    fig10 = px.histogram(
+        data_df, x="sepal.width", color="species", title="Histogram of Sepal width"
+    )
+    fig11 = px.histogram(
+        data_df, x="petal.length", color="species", title="Histogram of Petal length"
+    )
+    fig12 = px.histogram(
+        data_df, x="petal.width", color="species", title="Histogram of Petal width"
+    )
+    fig9.show()
+    fig10.show()
+    fig11.show()
+    fig12.show()
+    # Transforming the data using standard scaler
+    from sklearn.preprocessing import StandardScaler
+
+    scaler = StandardScaler()
+    x = scaler.fit_transform(x)
+    # Encoding the output variable
+    species = data_df.iloc[:, 4].values  # converting output variable to numpy array
+    from sklearn.preprocessing import LabelEncoder
+
+    encoder = LabelEncoder()
+    species = encoder.fit_transform(species)  # Encoding the out put values
+    # Splitting the data to test and training sets
+    from sklearn.model_selection import train_test_split
```

Comment:
I would suggest putting all imports at the start of the code.

Reply:
Oh, I thought to do this but missed that, thanks for mentioning this!